src/test/java/bio/terra/service/dataset/DatasetRequestValidatorTest.java

```diff
@@ -214,18 +214,41 @@ public void testInvalidKeyType() throws Exception {
     ColumnModel badColumnFileRefArray = testTable.getColumns().get(0);
     badColumnFileRefArray.setArrayOf(true);
     badColumnFileRefArray.setDatatype(TableDataType.FILEREF);
+    badColumnFileRefArray.setRequired(true);
```

Comment:
I am having a hard time following what this test is testing, and how the inputs translate to the expected outputs.  Can you help me understand?

Reply:
Yeah! So `testInvalidColumnMode` tests that a required array column causes an `InvalidColumnMode` error. `testInvalidKeyType` also causes this error because the primary key is an array